core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -235,8 +235,15 @@ public Pipeline get(String id) {
         return result;
     }
 
-    private Pipeline constructPipeline(String id, Map<String, Object> config) throws Exception {
-        return factory.create(id, config, processorFactoryRegistry);
+    WritePipelineResponse constructPipelineResponse(String id, Map<String, Object> config) {
```

Comment:
sure, I'll update to that. still feels like a weird method in general... but it works